### ISSUE TITLE
Performance on Seen Nodes lower than Matrix Factorization

### ISSUE THREAD

**Issue by ankitjain451** [williamleif/GraphSAGE#71]
Hey,
Thanks for the code. 

I have tried GraphSage approach on my dataset which is a bi-partite graph of two entities (for e.g. users and movies) . I am try to use the embeddings of nodes generated using GraphSage for link prediction task (what users will watch next).
I have an out of time dataset (which is not biased) on what users actually did (for example watched which movie).

I am trying to evaluate the performance of GraphSage and compare against the performance of standard matrix factorization on this task. However, what I have repeated found is following:
1. If users end of watching the same movie again in the validation set, Matrix Factorization performs way better than GraphSage in terms of Precision@K metric.
2. If users watch a movie which they have never watched, GraphSage performs better.

I have also consistently observed that GraphSage underfits the data (even after changing the samples_1 and samples_2 parameters). Is it by design or there is more to it?

**Comment by llan-ml** [comment]
Hi @ankitjain451 , can you explain more about "I have also consistently observed that GraphSage underfits the data".

I tested GraphSage on toy-ppi data, and set `--identity_dim 128` and do not use `topy-ppo_feats.npy`. The model completely overfit to training nodes (train_f1_mic=~1 after 60 epocs), and `val_f1_mic` always is 0.43.
